models/experimental/bert/fused_ops/linear.py

```diff
@@ -27,13 +28,11 @@ def Linear(
         assert bias.get_legacy_shape() == [1, 1, 32, out_features]
 
     def linear_(activation):
+        nonlocal bias
         weight_T = ttnn.transpose(weight, -2, -1)
-        output = ttnn.matmul(activation, weight_T)
-
-        if bias is not None:
-            output_plus_bias = tensor.bcast(output, bias, tensor.BcastOpMath.ADD, tensor.BcastOpDim.H)
-            return output_plus_bias
+        if bias is not None and bias.get_layout() != ttnn.TILE_LAYOUT:
+            bias = ttnn.to_layout(bias, ttnn.TILE_LAYOUT)
```

Comment:
Shouldn't this check/formatting of bias just be done before `linear_`? Then you don't need nonlocal/have this happen during runtime?

Reply:
By leaving the conversation of the bias within the _linear function it means the bias will be converted lazily and just-in-time.   This means that the calling function could do other things before the invocation.   Maybe I'm overthinking here though.

Reply:
I think most fns/classes we wrote for models aren't written in this JIT manner, and does the formatting when the fn/class is initialized. This is also a potentially minor perf hit we are paying for the layout check each time it is called, when we can avoid it by doing it outside during initialization.

Reply:
Ok, I will change it